File: 3289_the_two_sneaky_numbers_of_digitville/3289_the_two_sneaky_numbers_of_digitville.go
package main

func getSneakyNumbers(nums []int) []int {
	myMap := make(map[int]int)
	ans := make([]int, 2)
	j := 0

	for i := 0; i < len(nums); i++ {
		if val, ok := myMap[nums[i]]; ok {
			myMap[nums[i]] = val + 1
		} else {
			myMap[nums[i]] = 1
		}

		if myMap[nums[i]] == 2 {
			ans[j] = nums[i]
			j++
		}
	}

	return ans
}

File: 3289_the_two_sneaky_numbers_of_digitville/3289_the_two_sneaky_numbers_of_digitville.dart
void main(List<String> args) {
  print(getSneakyNumbers([0, 1, 1, 0]));
}

List<int> getSneakyNumbers(List<int> nums) {
  var myMap = Map<int, int>();
  List<int> ans = [];

  for (var i = 0; i < nums.length; i++) {
    if (myMap.containsKey(nums[i])) {
      myMap[nums[i]] = myMap[nums[i]]! + 1;
    } else {
      myMap.putIfAbsent(nums[i], () => 1);
    }

    if (myMap[nums[i]] == 2) {
      ans.add(nums[i]);
    }
  }

  return ans;
}
